components/disposable-focus/index.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * WordPress dependencies
+ */
+import { Component, findDOMNode } from 'element';
+
+function disposableFocus( WrappedComponent ) {
+	return class extends Component {
+		componentDidMount() {
+			this.activeElement = document.activeElement;
+		}
+
+		componentWillUnmount() {
+			const wrapper = findDOMNode( this );
```

Comment:
`findDOMNode` is [documented](https://facebook.github.io/react/docs/react-dom.html#finddomnode) as an escape hatch, though in this case I think a reasonable one. Concerns me to see [some conversation](https://twitter.com/dan_abramov/status/752936646602031104) in which React maintainers seem obviously inclined to put it on a path toward deprecation.

I noted the earlier point about wrapper components. I think we should keep this as-is, but leave open as options for future consideration:

- Wrap with `span` to reduce layout implications (inline, not block)
- Expect the mounted component to define a node getter (similar to [`react-click-outside`'s `handleClickOutside`](https://github.com/kentor/react-click-outside/blob/4934ef15531f3a0267e243aac39ed96dc10dfe45/index.js#L28))
- Pass callback as prop to the wrapped component, expected to be called on mount with reference to the DOM node